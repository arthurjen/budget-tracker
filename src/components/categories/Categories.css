:local(.categories) {
  width: 80%;
}