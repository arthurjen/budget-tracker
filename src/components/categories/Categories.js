import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import Category from './Category';
import { load } from './actions';
import { getCategories } from './reducers';
import styles from './Categories.css';

export class Categories extends PureComponent {

  static propTypes = {
    categories: PropTypes.array,
    load: PropTypes.func.isRequired
  };

  componentDidMount() {
    this.props.load();
  }

  render() { 
    const { categories } = this.props;
    if(!categories) return;
    
    return ( 
      <ul className={styles.categories}>
        {categories.map(category => (
          <Category
            key={category.id}
            category={category}
          />
        ))}
      </ul>
    );
  }
}
 
export default connect(
  state => ({ categories: getCategories(state) }),
  { load }
)(Categories);