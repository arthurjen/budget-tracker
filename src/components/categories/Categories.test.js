import React from 'react';
import { shallow } from 'enzyme';
import { Categories } from './Categories';
import toJSON from 'enzyme-to-json';

describe('Categories', () => {
  it('renders categories', () => {
    const categories = [
      { id: '1', name: 'a' },
      { id: '2', name: 'b' },
      { id: '3', name: 'c' }
    ];
    const load = jest.fn();
    const wrapper = shallow(
      <Categories categories={categories} load={load} />
    );

    expect(toJSON(wrapper)).toMatchSnapshot();
  });

});