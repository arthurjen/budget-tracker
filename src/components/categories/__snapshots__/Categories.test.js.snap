// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Categories renders categories 1`] = `
<ul
  className="categories"
>
  <Connect(Category)
    category={
      Object {
        "id": "1",
        "name": "a",
      }
    }
    key="1"
  />
  <Connect(Category)
    category={
      Object {
        "id": "2",
        "name": "b",
      }
    }
    key="2"
  />
  <Connect(Category)
    category={
      Object {
        "id": "3",
        "name": "c",
      }
    }
    key="3"
  />
</ul>
`;
